### ISSUE TITLE
Block search input alignment doesn't seem right

### ISSUE THREAD

**Issue by towerofnix** [jmoenig/Snap--Build-Your-Own-Blocks#1865]
When we're open in the "find blocks" palette, and the palette is at normal width, the magnifying glass icon overlaps the search field:

![Overlap screenshot](https://user-images.githubusercontent.com/9948030/30693878-d978fd44-9ea7-11e7-9c90-6d1e3daf9788.png)

I think it would be better if it automatically stretched the search input so that there's a few pixels of margin between the magnifying glass button and the right edge of the search input.

Plus, that would solve this issue: at the moment, the search field doesn't resize based on your palette width, and I think it should.

![Resized palette screenshot](https://user-images.githubusercontent.com/9948030/30693942-24ad9d1a-9ea8-11e7-91fe-125b7c1b318c.png)

**Comment by brianharvey** [comment]
Or when the search bar is shown, don't show the magnifying glass at all!

**Comment by towerofnix** [comment]
I think the idea is that the magnifying glass is an easy way to get in *and* out of the search bar. If the user clicks on the magnifying glass, they shouldn't have to use the escape key - which is on the keyboard, not the mouse - to get out of there. (To be fair, they could just select a different block category, but this doesn't seem like it would be obvious.)

**Comment by brianharvey** [comment]
Oh!  It didn't even occur to me that clicking it again would get you out.

**Comment by jmoenig** [comment]
hmmm... that's what I did at first, not show the icon in the search pane at all. Then, on second thought, I added the button to users can click it to get back. But you're right, that might be overenginerring it.

**Comment by cycomachead** [comment]
Just curious -- why this button right here? I thought there was an objection to adding buttons like this? I think is good for visibility, though so I don't mind.

However, thinking about this location, a `+` for "Make a Block" would probably be more useful.

**Comment by jmoenig** [comment]
Yeah, I'm coming around ;-)
The idea is to make some of the really cool Snap! features more discoverable, including search (what this is about), make a block (which I'm thinking about adding a block to each palette at the bottom of the primitives that will pre-select the category's color in the "make a block" dialog, and also a toggle in the top toolbar for "visible stepping".

**Comment by cycomachead** [comment]
Hahaha. Well there’s a closed PR that already has the code for the male a block button. :)

That seems fine by me too. Search being easily findable will be used by many, so I’m still glad to see this. 

-- 
Michael Ball
From my iPhone
michaelball.co

**Comment by brianharvey** [comment]
I like the idea of an always-visible make-a-block button.  There's a huge chunk of unused real estate to the right of the sprite name, up top.  The bottom of a palette _isn't_ always visible.
No need for a toggle for visible stepping!  Since the left end of the slider already means something special (single step), the right end should too (no stepping).  You could even make that more obvious by having the slider light up except when it's all the way to the right.
And while we're chatting about buttons, I still hate it that after you push the undo button it turns into the redo button, and the undo button appears somewhere else.  That's really counterintuitive.  Make the redo button always visible, but greyed out if there's nothing to redo.

**Comment by towerofnix** [comment]
> And while we're chatting about buttons, I still hate it that after you push the undo button it turns into the redo button, and the undo button appears somewhere else. That's really counterintuitive. Make the redo button always visible, but greyed out if there's nothing to redo.

+1!

(I'd add a :+1: to your post but you made multiple points, and it'd be ambiguous which one I'm particularly agreeing with.)

**Comment by cycomachead** [comment]
I agree with BH about the always visible-ness I’d make a block, though I do think it should be connected to the palette a bit more. 

Not sure what’s best. 

-- 
Michael Ball
From my iPhone
michaelball.co
